oauthenticator/vsteamservices.py

```diff
@@ -0,0 +1,110 @@
+"""
+Custom Authenticator to use Visual Studio Team Service OAuth with JupyterHub
+
+"""
+
+import json
+import os
+import urllib
+
+from jupyterhub.auth import LocalAuthenticator
+from tornado import gen
+from tornado.auth import OAuth2Mixin
+from tornado.httpclient import HTTPRequest, AsyncHTTPClient
+from tornado.httputil import url_concat
+
+from .oauth2 import OAuthLoginHandler, OAuthenticator
+
+VSTS_HOST = os.environ.get('VSTS_HOST') or 'app.vssps.visualstudio.com'
+
+
+class VstsEnvMixin(OAuth2Mixin):
+    _OAUTH_AUTHORIZE_URL = "https://%s/oauth2/authorize" % VSTS_HOST
+    _OAUTH_ACCESS_TOKEN_URL = "https://%s/oauth2/token" % VSTS_HOST
+
+
+class VstsLoginHandler(OAuthLoginHandler, VstsEnvMixin):
```

Comment:
Minor style nit: Would you mind using `VSTS` instead of `Vsts` for the acronym part?